docs: add user-documentation and consolidate various docs

bilalshaikh42 opened the pull request:
@jonrkarr  I have also added the information about the conventions here. I think its much easier to maintain markdown vs the angular app,  and provides a clearner user experience, so if you agree we can remove the docs from the angular apps

bilalshaikh42 committed:
docs: add user guide

bilalshaikh42 committed:
docs: add concepts/conventions to docs

bilalshaikh42 committed:
docs: add author-docs

bilalshaikh42 committed:
chore: fix spellings

jonrkarr commented on docs/stylesheets/theme.css:
```diff
@@ -7,7 +7,7 @@
   --md-accent-fg-color--light: #ffe0b2;
   --md-accent-fg-color--dark: #ff7b00;
-  --md-text-font-family: 'Raleway';
+  --md-text-font-family: 'Roboto';
```
The main font is Raleway

jonrkarr commented on docs/users/user-guide.md:
```diff
@@ -2,15 +2,59 @@
+
+### Searching for projects
+![search-projects](./images/search.png){align=right}
+Clicking on the the search icon at the top right of the page opens a search box. A search term, such as 'metabolism' can be entered in the search box. By default, the search term is searched against each [attribute](#selecting-attributes) of the project. Optionally, you can restrict the search to a specific attribute. For example, if you want to search for projects that have the value 'Eschiria c ...
```
"Eschiria" is spelled "Escherichia"

The dashes should be used in the attribute names. A list of the attributes is in the FAQs page.

I'm not sure that this the lunr search syntax for multiword phrases. There doesn't appear to be [documentation](https://lunrjs.com/guides/searching.html) for this.

jonrkarr commented on docs/users/user-guide.md:
```diff
@@ -2,15 +2,59 @@
 
-### Configuring new visualizations
+Projects can be visualized using both predefined and custom visualizations. The "Select Chart" tab allows you to select from visualizations that are defined and included in the project. Additionally, you can create your own custom visualization, by selecting one of  the "Design a chart" options. These include histograms, heatmaps and lineplots. Selecting a a plot time will open an additional menu ...
+Once you have configured your visualization, you can view it by clicking on the "View chart" button.
```
typo "a a"

jonrkarr commented on docs/users/user-guide.md:
```diff
@@ -2,15 +2,59 @@
-### Configuring new visualizations
+Projects can be visualized using both predefined and custom visualizations. The "Select Chart" tab allows you to select from visualizations that are defined and included in the project. Additionally, you can create your own custom visualization, by selecting one of  the "Design a chart" options. These include histograms, heatmaps and lineplots. Selecting a a plot time will open an additional menu ...
+Once you have configured your visualization, you can view it by clicking on the "View chart" button.
+
```
Add information about exporting visualizations to Vega

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
+
+One key feature of Vega is that it modularly captures the graphical marks which comprise visualizations and how those graphical marks should be painted with data. This feature makes it easy to produce data visualizations for multiple simulation conditions by combining the same graphical marks with results from multiple simulations. This features also makes the provenance of data visualizations tr ...
+
+Vega files can be included in the ComBINE/OMEX archive of a simulation project to enable visualization on the [project view page](./user-guide.md#visualizations).
```
"ComBINE" is spelled "COMBINE"

bilalshaikh42 replied:
Not sure how that happened, thanks

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
+ 
 ## Validating projects
 
+BioSimulations provides a validation tool for validating the contents of a COMBINE/OMEX archive. This tool is available on [runBioSimulations](https://run.biosimulations.org/utils/validate-project). The user can upload a COMBINE/OMEX archive or provide a URL to a COMBINE/OMEX archive. The tool will validate the contents of the archive and report any errors. The interface provides a customizable r ...
```
This report is not customizable by users.

jonrkarr replied:
"Sed-ML" is spelled "SED-ML"

bilalshaikh42 replied:
By customizable i was referring to the various options on selecting which parts of the project to validate. I can see how this is confusing. I'll reword this

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
+
 ## Using BioSimulators to test projects prior to submission to BioSimulations
 
+Authors can run their simulation projects locally using the BioSimulators docker images. To run the project locally, pull the appropriate BioSimulators docker image, and then run the simulator as follows:
```
"docker" is spelled "Docker"

jonrkarr replied:
Before this I think its helpful to communicate that BioSimulations uses BioSimulators to execute simulations. Together with this paragraph, this communicates that authors can execute simulations just as BioSimulations does, using the same interfaces to the same tools.

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
 # Author guide: creating and publishing projects
 
-## Creating projects
+To make it easy for investigators to work with a broad range of model formats, modeling frameworks, simulation types, simulation algorithms, and simulation tools, BioSimulations uses existing community standards for creating and describing projects. BioSimulations uses the Simulation Experiment Description Markup Language (SED-ML) to describe [simulation experiments](#sed-ml) and COMBINE/OMEX arc ...
```
I think it would be helpful to start with a more explicit overview of the publication process:
1. Create a simulation project (COMBINE/OMEX archive)
   - One or more models (e.g., BNGL, CellML, SBML)
   - One or more simulations (SED-ML)
   - Zero or more visualizations of the results of the simulations
   - Metadata about the project (e.g., taxa, citations, license)
2. Execute the project using runBioSimulations
3. Use runBioSimulations to verify that the project executed as intended
4. Use runBioSimulations to publish the project

jonrkarr replied:
The description of conventions could follow the overview as a separate subsection, "Conventions for simulation projects"

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
+To make it easy for investigators to work with a broad range of model formats, modeling frameworks, simulation types, simulation algorithms, and simulation tools, BioSimulations uses existing community standards for creating and describing projects. BioSimulations uses the Simulation Experiment Description Markup Language (SED-ML) to describe [simulation experiments](#sed-ml) and COMBINE/OMEX arc ...
 
-### SED-ML
+## Creating Projects
```
Title capitalization should be "Creating projects", consistent with other sections

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
+1. The simulation runs successfully and all outputs are present.
+1. The simulation project metadata is accurate and complete.
+    1. The thumbnail image is loaded and displayed correctly.
+    1. The simulation project has a title, descriptions, and information about the authors
```
The end-punctuation of the list items should be consistent

jonrkarr replied:
"descriptions" should be singular

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
 
+Before publishing a simulation project, the user should review the simulation results. The author should verify the following: 
+
+1. The simulation runs successfully and all outputs are present.
```
I think this order would be easier to follow:

- Project includes all intended files
- Project executed successfully
- Outputs are as intended
- Metadata

bilalshaikh42 replied:
Agreedm thanks

jonrkarr commented on docs/users/author-guide.md:
```diff
@@ -1,28 +1,106 @@
+No login is required to access BioSimulations. However, users must have an account to manage projects. This allows for proper crediting of authors and allows authors to manage and edit their projects. BioSimulations provides a free account for users who wish to publish and manage projects.
 
+!!!note
+    User accounts are under development. If you are interested in submitting a project to BioSimulations, please contact us at info@biosimulations.org
```
End sentence with period.

jonrkarr commented on docs/users/user-guide.md:
```diff
@@ -2,15 +2,59 @@
 
 ## Finding projects
 
+Published projects can be browsed at [https://biosimulations.org/projects](https://biosimulations.org/projects). Each card presents a project, with a thumbnail, and title. Mousing over the thumbnail shows additional details about the project. You can customize the attributes you want to view. 
```
Remove extra second comma in "each card presents a project, with a thumbnail, and title"

bilalshaikh42 committed:
docs: change font to raleway

bilalshaikh42 committed:
docs: add snippets to track supported standards

added snippets to describe supported langauges and sedml features consistently throughout docs

bilalshaikh42 commented:
@jonrkarr could you take a look the two snippets I added for the supported languages and sedml features? I have not added the links to external websites yet but want to confirm that the info is correct and current.

jonrkarr committed:
docs: clarified SED-ML support from tools

jonrkarr committed:
docs: clarified model languages

jonrkarr committed:
docs: clarified user guide

jonrkarr committed:
docs: clarified author guide

jonrkarr commented:
I edited some of the files. Merge when you're ready. I can review again when the docs are complete.

bilalshaikh42 committed:
docs: spelling corrections

bilalshaikh42 committed:
chore: add words to spelling list

bilalshaikh42 committed:
docs: add in runBioSimulations and BioSimulators tutorials
consolidate the tutorial pages for all apps into one user guide

bilalshaikh42 commented:
@jonrkarr I added in the resources from BioSimulators and runBioSimulations as well to the user guide. Does the FAQ contain everything already? 
Let me know if there is something that I missed. Some of the information was repeated across the apps so I may have missed copying something over.

bilalshaikh42 committed:
docs: spelling fixes

bilalshaikh42 committed:
docs: add spelling ignore pattersn for urls

bilalshaikh42 committed:
docs: spelling fixes

biosimulations-daemon commented:
:tada: This PR is included in version 8.0.0 :tada:

The release is available on [GitHub release](https://github.com/biosimulations/biosimulations/releases/tag/v8.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
